### ISSUE TITLE
Composer is partially hidden on small mobile screens when keyboard is open

### ISSUE THREAD

**Issue by matteocontrini** [flarum/core#2628]
## Bug Report

**Current Behavior**

After #1670 was fixed, the composer is now fixed at the top of the screen on mobile. On small screens, this means that the keyboard could cover the composer and what you're trying to write.

This was also discussed in the PR (#2279), but I think the consequences were underestimated, because the new behaviour is a serious regression in the usability of the composer. Watch the video below to feel the frustration :)

**Steps to Reproduce**
1. Go to nightly.flarum.site with a "small" screen (16:9 is probably enough)
2. Attempt to write a long post
3. Observe that you can't see what you're typing

**Expected Behavior**

There should definitely be a way to see what you're typing. Previously, it was possible to scroll to move the composer.

**Screenshots**

This video was provided by a user of my forum, where I cherry-picked and deployed the commit 9b27b0d9d737931b8b7d93769c1e69bcc1ef6b6a.

https://user-images.githubusercontent.com/2164763/108773766-22bbad80-755f-11eb-92d3-b82a2c718ed1.mp4

**Environment**
- Flarum version: any version after 9b27b0d9d737931b8b7d93769c1e69bcc1ef6b6a

**Comment by SychO9** [comment]
Thank you for reporting this!

@matteocontrini could you please go to nighly right now (before it's reset) and see if the issue still occurs ?

**Comment by matteocontrini** [comment]
@SychO9 thanks for the fast reply :)

I asked my user with the small screen and he says he's not experiencing the issue on nightly.

**Comment by SychO9** [comment]
Great!

I'll push the fix later, thank you.

**Comment by matteocontrini** [comment]
There's an additional problem caused by the fixed position of the modal, and it's the fact that you cannot scroll the mentions dropdown. (Feel free to skip the first 15 seconds of the video.)


https://user-images.githubusercontent.com/2164763/109331130-c1485700-785c-11eb-980b-2d3e76e3d669.mp4

**Comment by askvortsov1** [comment]
Ooo, also the buttons for emoji/mentions are cut off. That's not ideal...
On the bright side, the mentions dropdown should no longer go above the top of the screen as part of the positioning changes associated with #2594

**Comment by SychO9** [comment]
> Ooo, also the buttons for emoji/mentions are cut off. That's not ideal...

I believe that area is scrollable so it shouldn't be a problem

**Comment by matteocontrini** [comment]
New problem reported by an iOS user, and he says it happens very often.

(Changes in #2631 were deployed to the forum some days ago.)

![ezgif-com-video-to-gif](https://user-images.githubusercontent.com/2164763/109642259-05847180-7b53-11eb-83b6-469d489ba491.gif)

**Comment by SychO9** [comment]
Are both https://github.com/flarum/core/pull/2279 and https://github.com/flarum/core/pull/2631 applied ?

**Comment by matteocontrini** [comment]
@SychO9 yes.

Here's [another GIF](https://i.ibb.co/K0KZ4Dc/ezgif-com-video-to-gif.gif). He says that it happens "almost every time".

**Comment by SychO9** [comment]
That's very odd, how come he is scrolling to the composer, when the composer is no longer absolute, it's as if https://github.com/flarum/core/pull/2279 is not applied, but it clearly is. I guess it's one of those weird iOS behaviours.

Could you please create a new issue about it until we can reproduce it and find a fix ?

**Comment by matteocontrini** [comment]
Opened #2652.
